### PR TITLE
feat: semantic versioning

### PR DESCRIPTION
**Pull request by tomwwinter**

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Dockerfile
+.dockerignore
```
Comment (sleidig) [inline .dockerignore:2]: why do we ignore the Dockerfile?

File: .github/workflows/pull-request-update.yml [hunk 2]
```diff
@@ -1,28 +0,0 @@
-name: Pipeline
-on: pull_request
```
Comment (sleidig) [inline .github/workflows/pull-request-update.yml:2]: should we still test to build on PRs?

File: .github/workflows/build-release.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      packages: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - name: Login to GitHub Container Registry
```
Comment (sleidig) [inline .github/workflows/build-release.yml:23]: If we switch to GitHub Container Registry from DockerHub, let's check all docs (espcially ndb-setup repo) to update links.

**Review by sleidig — APPROVED** [review]

File: .dockerignore (hunk 1 above)
Comment (tomwwinter) [reply .dockerignore:2]: It just means, that we will not copy the Dockerfile into the Container.

File: .github/workflows/pull-request-update.yml (hunk 2 above)
Comment (tomwwinter) [reply .github/workflows/pull-request-update.yml:2]: The test just moved to another workflow, we still run the test on PRs

File: .github/workflows/build-release.yml (hunk 3 above)
Comment (tomwwinter) [reply .github/workflows/build-release.yml:23]: Yes, i will update all the docs.

**Pull request merged**